automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/BatchNodepoolCreationSpec.scala

```diff
@@ -0,0 +1,144 @@
+          x => x.map(_.status).distinct == List(AppStatus.Deleted)
+
+        for {
+          clusterName <- IO.fromEither(KubernetesNameUtils.getUniqueName(KubernetesClusterName.apply))
+          _ <- LeonardoApiClient.batchNodepoolCreate(
+            googleProject,
+            defaultBatchNodepoolRequest.copy(clusterName = Some(clusterName), numNodepools = NumNodepools(1))
+          )
+          getCluster = gkeServiceResource.use { gkeClient =>
+            val id = KubernetesClusterId(googleProject, LeonardoConfig.Leonardo.location, clusterName)
+            gkeClient.getCluster(id)
+          }
+
+          monitorBatchCreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getCluster, 20, 10 seconds)(
+            testTimer,
+            clusterDoneCheckable
+          ).compile.lastOrError
+
+          //here we sleep, because the above verifies the google state and we need to wait until leo has polled and updated its internal state to proceed
+          //it is a long time because there is a lot of stuff leo has to do besides GKE entity creation
+          _ <- testTimer.sleep(5 minutes)
+
+          _ = monitorBatchCreationResult.map(_.getNodePoolsList().size()) shouldBe Some(2)
+
+          appName1 = AppName("app1")
+          diskConfig1 = Some(PersistentDiskRequest(DiskName("disk1"), None, None, Map.empty))
+          _ <- LeonardoApiClient.createApp(googleProject,
+                                           appName1,
+                                           createAppRequest = defaultCreateAppRequest.copy(diskConfig = diskConfig1))
+
+          getApp1 = LeonardoApiClient.getApp(googleProject, appName1)
+          monitorApp1CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp1, 20, 10 seconds)(
+            testTimer,
+            appDoneCheckable
+          ).compile.lastOrError
+
+          clusterAfterApp1 <- getCluster
+          _ = clusterAfterApp1.map(_.getNodePoolsList().size()) shouldBe Some(2)
+
+          appName2 = AppName("app2")
+          diskConfig2 = Some(PersistentDiskRequest(DiskName("disk2"), None, None, Map.empty))
+          _ <- LeonardoApiClient.createApp(googleProject,
+                                           appName2,
+                                           createAppRequest = defaultCreateAppRequest.copy(diskConfig = diskConfig2))
+
+          //creating a second app with 1 precreated nodepool should cause a second user nodepool to be created
+          getApp2 = LeonardoApiClient.getApp(googleProject, appName2)
+          monitorApp2CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp2, 20, 10 seconds)(
+            testTimer,
+            appDoneCheckable
+          ).compile.lastOrError
+
+          clusterAfterApp2 <- getCluster
+          _ = clusterAfterApp2.map(_.getNodePoolsList().size()) shouldBe Some(3)
+
+          _ <- LeonardoApiClient.deleteApp(googleProject, appName1)
```

Comment:
we need to do this so Leo cleans up the App SAM record, otherwise Gpalloc cleanup gets mad.